httpx/_content.py

```diff
@@ -61,6 +61,21 @@ async def __aiter__(self) -> AsyncIterator[bytes]:
             yield part
 
 
+class UnattachedStream(AsyncByteStream, SyncByteStream):
+    """
+    If a request or response is serialized using pickle, then it is no longer
+    attached to a stream for I/O purposes. Any stream operations should result
+    in `httpx.StreamClosed`.
+    """
+
+    def __iter__(self) -> Iterator[bytes]:
+        raise ResponseClosed()  # TODO: StreamClosed
+
+    async def __aiter__(self) -> AsyncIterator[bytes]:
+        raise ResponseClosed()  # TODO: StreamClosed
+        yield b""  # pragma: nocover
```

Comment:
We need the `yield` here to avoid `TypeError`:

```
    async def aread(self) -> bytes:
        """
        Read and return the request content.
        """
        if not hasattr(self, "_content"):
            assert isinstance(self.stream, typing.AsyncIterable)
>           self._content = b"".join([part async for part in self.stream])
E           TypeError: 'async for' received an object from __aiter__ that does not implement __anext__: coroutine
```